### PR TITLE
Hardening of shell and enable rubocop

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/clients/kdump.rb [hunk 1]
```diff
@@ -1206,12 +1197,11 @@ def cmdKdumpSMTPUser(options)
...
     end
 
-
     def cmdKdumpSMTPPass(options)
       options = deep_copy(options)
-      if Ops.get(options, "pass") != nil
+      if Ops.get(options, "pass")
         password = cmdParsePassPath(Builtins.tostring(Ops.get(options, "pass")))
-        return false if password == nil || password == ""
+        return false if password.nil? || password == ""
```
Comment (dgdavid) [inline src/clients/kdump.rb]: You also can use `if password.to_s.empty?`

File: src/modules/Kdump.rb [hunk 2]
```diff
@@ -1137,27 +1124,18 @@ def crash_kernel_values
...
-      if low && low.to_i != 0
-        # Add the ',low' suffix only there is a ',high' one
-        if result.empty?
-          result << "#{low}M"
-        else
-          result << "#{low}M,low"
-        end
-      end
+      # Add the ',low' suffix only there is a ',high' one
+      result << (result.empty? ? "#{low}M" : "#{low}M,low") if low && low.to_i != 0
```
Comment (dgdavid) [inline src/modules/Kdump.rb]: Quite complex. Sincerely, I would split it

```ruby
...
low_prefix = result.empty? ? "#{low}M" : "#{low}M,low"
result << low_prefix if low && !ow.to_i != 0Ç
...
```

File: src/modules/Kdump.rb [hunk 3]
```diff
@@ -1214,11 +1190,7 @@ def remove_offsets!(values)
     def write_fadump_boot_param
       if system.supports_fadump?
         # If fdump is selected and we want to enable kdump
-        if using_fadump? && @add_crashkernel_param
-            value = "on"
-        else
-            value = nil
-        end
+        value = ("on" if using_fadump? && @add_crashkernel_param)
```
Comment (dgdavid) [inline src/modules/Kdump.rb]: Just a question: are the parenthesis really needed? Why?

File: test/client_test.rb [hunk 4]
```diff
@@ -1,50 +1,50 @@
...
     context "when using only alloc memory low" do
-      let(:options) {{"enable"=>"", "alloc_mem"=>"#{alloc_mem_low}"}}
-      
+      let(:options) { { "enable" => "", "alloc_mem" => alloc_mem_low.to_s } }
+
       it "enables only the low value of Kdump allocated_memory " do
         subject.cmdKdumpStartup(options)
-        expect(Yast::Kdump.allocated_memory).to eq({low: alloc_mem_low,
-          high: nil})
+        expect(Yast::Kdump.allocated_memory).to eq(low:  alloc_mem_low,
```
Comment (dgdavid) [inline test/client_test.rb]: NP: Does not fit in the same line?

File: test/client_test.rb [hunk 5]
```diff
@@ -1,50 +1,50 @@
...
     context "When using alloc memory high and low" do
-      let(:options) {{"enable"=>"", "alloc_mem"=>"#{alloc_mem_low},#{alloc_mem_high}"}}
-    
+      let(:options) { { "enable" => "", "alloc_mem" => "#{alloc_mem_low},#{alloc_mem_high}" } }
+
       it "sets the low and high values of  Kdump allocated_memory " do
         subject.cmdKdumpStartup(options)
-        expect(Yast::Kdump.allocated_memory).to eq({low: alloc_mem_low,
-          high: alloc_mem_high})
+        expect(Yast::Kdump.allocated_memory).to eq(low:  alloc_mem_low,
```
Comment (dgdavid) [inline test/client_test.rb]: NP: Does not fit in the same line?

File: src/modules/Kdump.rb (hunk 2 above)
Comment (jreidinger) [reply src/modules/Kdump.rb]: this way on other hand you use `low` which can be nil, which does not crash, but is a bit strange.

File: src/modules/Kdump.rb (hunk 3 above)
Comment (jreidinger) [reply src/modules/Kdump.rb]: no, this should not be needed.

File: test/client_test.rb (hunk 4 above)
Comment (jreidinger) [reply test/client_test.rb]: let me check.

**Pull request merged**